src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts

```diff
@@ -0,0 +1,306 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as DOM from 'vs/base/browser/dom';
+import { Disposable, DisposableStore } from 'vs/base/common/lifecycle';
+import { INotebookEditor } from 'vs/workbench/contrib/notebook/browser/notebookBrowser';
+import { INotebookCellList } from 'vs/workbench/contrib/notebook/browser/view/notebookRenderingCommon';
+import { NotebookCellOutlineProvider, OutlineEntry } from 'vs/workbench/contrib/notebook/browser/viewModel/notebookOutlineProvider';
+import { CellKind } from 'vs/workbench/contrib/notebook/common/notebookCommon';
+
+
+export class NotebookEditorStickyScroll extends Disposable {
+	private readonly _disposables = new DisposableStore();
+
+	constructor(
+		private readonly domNode: HTMLElement,
+		private readonly notebookEditor: INotebookEditor,
+		private readonly notebookOutline: NotebookCellOutlineProvider,
+		private readonly notebookCellList: INotebookCellList
+	) {
+		super();
+
+		if (this.notebookEditor.notebookOptions.getLayoutConfiguration().stickyScroll) {
+			this.init();
+		}
+
+		this._register(this.notebookEditor.notebookOptions.onDidChangeOptions((e) => {
+			if (e.stickyScroll) {
+				this.updateConfig();
+			}
+			if (e.globalToolbar) {
+				this.setTop();
+			}
+		}));
+	}
+
+	private updateConfig() {
+		if (this.notebookEditor.notebookOptions.getLayoutConfiguration().stickyScroll) {
+			this.init();
+		} else {
+			this._disposables.clear();
+			DOM.clearNode(this.domNode);
+			this.updateDisplay();
+		}
+	}
+
+	private setTop() {
+		if (this.notebookEditor.notebookOptions.getLayoutConfiguration().globalToolbar) {
```

Comment:
`this.notebookEditor.notebookOptions.getLayoutConfiguration().globalToolbarHeight`